libraries/AP_HAL_Linux/UARTDriver.cpp

```diff
@@ -450,12 +353,11 @@ bool UARTDriver::_write_pending_bytes(void)
     uint16_t n;
 
     // write any pending bytes
-    uint16_t _tail;
-    uint16_t available_bytes = BUF_AVAILABLE(_writebuf);
+    uint32_t available_bytes = _writebuf.available();
     n = available_bytes;
     if (_packetise && n > 0 &&
-        (_writebuf[_writebuf_head] != MAVLINK_STX_MAVLINK1 &&
-         _writebuf[_writebuf_head] != MAVLINK_STX)) {
+        (_writebuf.peek(0) != MAVLINK_STX_MAVLINK1 &&
+         _writebuf.peek(0) != MAVLINK_STX)) {
```

Comment:
Not a blocker, but this should be changed to the same as below, saving it in a variable.